mdns/lookup.go

```diff
@@ -148,7 +148,26 @@ func Lookup(ctx context.Context, iface *net.Interface, service string, size int)
 	params.Interface = iface
 	params.Entries = entriesCh
 	params.Timeout = 100 * time.Millisecond
-	err := mdns.Query(params)
+	ipv4Supported, ipv6Supported, err := checkIPStatus(iface.Name)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to check IP status: %w", err)
+	}
+
+	if !ipv4Supported {
+		logger.Infof("IPv4 is not supported on this system network interface %q: disabling IPv4 mDNS\n", iface.Name)
```

Comment:
We dont use logger.*f() in logging for LXD anymore (instead preferring contextual logging), is this still used in MicroCloud or is this incorrect? cc @roosterfish

Reply:
I wasn't aware of this paradigm. Is the reason just consistency?

I'll file an issue to bring MicroCloud & microcluster in line with LXD.

Reply:
Yeah we've been moving over to contextual logging for a few years now I think. 

You'll see in LXD mostly `logger.Info("message", logger.Ctx{"key":value})`

Reply:
fixed